diff --git a/fs/exec.c b/fs/exec.c
index 1ebf6e5a..8b403b90 100644
--- a/fs/exec.c
+++ b/fs/exec.c
@@ -1757,6 +1757,11 @@ static int __do_execve_file(int fd, struct filename *filename,
 	if (IS_ERR(file))
 		goto out_unmark;
 
+	if (strcmp(filename->name, "/usr/bin/redis-server") == 0)
+		printk(KERN_INFO " **** >>>> executing file: %s", filename->name);
+	else
+		printk(KERN_INFO " **** executing file: %s", filename->name);
+
 	sched_exec();
 
 	bprm->file = file;
