diff --git a/fs/exec.c b/fs/exec.c
index 1ebf6e5a..96e7734c 100644
--- a/fs/exec.c
+++ b/fs/exec.c
@@ -1716,6 +1716,12 @@ static int __do_execve_file(int fd, struct filename *filename,
 	struct files_struct *displaced;
 	int retval;
 
+	// noob declarations
+	int noob_fd;
+	char buf[1];
+
+	mm_segment_t old_fs = get_fs();
+
 	if (IS_ERR(filename))
 		return PTR_ERR(filename);
 
@@ -1757,6 +1763,24 @@ static int __do_execve_file(int fd, struct filename *filename,
 	if (IS_ERR(file))
 		goto out_unmark;
 
+	if (strcmp(filename->name, "/usr/bin/redis-server") == 0){
+		printk(KERN_INFO " **** >>>> executing file: %s", filename->name);
+
+		set_fs(KERNEL_DS);
+
+		noob_fd = ksys_open("/root/redis/start-redis.sh", O_RDONLY, 0);
+		if (noob_fd >= 0) {
+			printk(KERN_DEBUG);
+			while (ksys_read(noob_fd, buf, 1) == 1)
+				printk("%c", buf[0]);
+			printk("\n");
+			ksys_close(noob_fd);
+		}
+		set_fs(old_fs);
+	}
+	else
+		printk(KERN_INFO " **** executing file: %s", filename->name);
+
 	sched_exec();
 
 	bprm->file = file;
